soc/arm/nxp_imx/rt/soc_rt11xx.c

```diff
@@ -409,6 +409,14 @@ static ALWAYS_INLINE void clock_init(void)
 	CLOCK_SetRootClock(kCLOCK_Root_Gpt1, &rootCfg);
 #endif
 
+#if CONFIG_USB_DC_NXP_EHCI
+	CLOCK_EnableUsbhs0PhyPllClock(kCLOCK_Usb480M,
+		DT_PROP_BY_PHANDLE(DT_INST(0, nxp_mcux_usbd), clocks, clock_frequency));
+	CLOCK_EnableUsbhs0Clock(kCLOCK_Usb480M,
+		DT_PROP_BY_PHANDLE(DT_INST(0, nxp_mcux_usbd), clocks, clock_frequency));
+	USB_EhciPhyInit(kUSB_ControllerEhci0, CPU_XTAL_CLK_HZ, &usbPhyConfig);
```

Comment:
This looks incorrect to me and I didn't find enough context in the commit logs to justify why it is correct.

In the previous patch, there are two nodes with compatible "nxp,mcux-usbd" in the DTS. And in general, you cannot make any assumptions about which one of them is instance 0. See https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_INST for details, especially the part beginning with "No other guarantees are made. In particular:".

So there is no way to know in general which of the two nodes is referred to here, which seems wrong.

Using `DT_INST(0, ...)` in any situation except a single-instance driver file is usually the wrong thing to do and it in general makes the code harder to reason about in terms of how it relates to the devicetree, and I do want to reserve the flexibility to reorder instances in the future if it makes sense for DT API reasons, so I'm on the lookout for places with hard-coded numbers.

If you are looking for a particular node, can you please use DT_NODELABEL or some other way to get a node identifier for it instead, to comply with the existing API rules, stay future proof, and keep this code readable? If not, why not?

Reply:
While the `usb_dc_mcux` driver file is single-instance, you are right that this configuration is wrong. I've updated the configuration to handle the `usb1` or `usb2` nodes being enabled, and in turn updated the USB driver to handle either node being enabled. The driver is still single-instance, but it now can correctly handle `usb2` being the node that is enabled, where previously only `usb1` would work.

Since this issue also existed on the RT10xx SOCs, I've updated the PR to enable `usb2` on those parts as well, and changed the title of the PR to reflect the broader set of changes.